### ISSUE TITLE
Atominus by Maza (Protracker) sounds off

### ISSUE THREAD

**Issue by blaqmajik** [steffest/BassoonTracker#22]
The lead coming in at position 07 remains 'stale' and it seems it's not re-triggered in the same way it is when you use ProTracker - the alternating "0a, 0b, 0c, 0d, 0e"s in the sample/instrument column should switch to the respective sample, even when there is no new note triggered.

**Comment by 8bitbubsy** [comment]
This is called sample swapping and is not trivial to implement correctly. Many trackers didn't even support it outside of NoiseTracker/ProTracker.

**Comment by steffest** [comment]
Yes, that "on the fly sample swapping" thing is an annoying one.
That's one of these things that wat there in Protracker 1 and 2, but not anymore in Protracker 3 and beyond. (And also not in Fasttracker)
I'm still pondering if I should implement it ... I tend to lean to "no" - 100% compatiblity is not really the goal - or even achievable ... that's only something @8bitbubsy can do!

**Comment by 8bitbubsy** [comment]
> 
> 
> 100% compatiblity is not really the goal - or even achievable ... that's something only @8bitbubsy can do!

Hehe, the only reason I can get it 100% right is because I'm porting original replayers instead of writing my own. It's a challenge to get these old quirks right when you roll your own replayer, that's for sure...

**Comment by blaqmajik** [comment]
Fair point. It is used in a lot of chiptunes, though. 

And not just in obscure ones but in absolute crack intro classics like "Digital Bass-Line" from Emax, for example.

**Comment by bryc** [comment]
Sample swapping works in NT1/2 and PT1/2. but _doesn't_ work in PT3 or FT1/2. I used to use that trick but went back to manually put in notes where they should be for maximum player compatibility. Not like you are saving any bytes by omitting them..

**Comment by 8bitbubsy** [comment]
Yep, PT3 is known for breaking sample swapping. Did you know that sample swapping actually works in Sound Blaster mode in Scream Tracker 3? Little known fact. It does however not work the same way as ProTracker (where it will switch to the new sample after one cycle of the current sample has been done).
